### PR TITLE
Fix types in react-gallery-carousel

### PR DESCRIPTION
**Pull request by guitheengineer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/yifaneye/react-gallery-carousel/issues/69#issuecomment-1037108269
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/react-gallery-carousel/index.d.ts [hunk 1]
```diff
@@ -64,12 +64,7 @@ declare namespace Carousel {
         children?: React.ReactNode;
     }
 
-    // Buttons of the carousel that accepts a boolean or string.
-    type ButtonType = boolean | string;
-
-    // Object fit css properties
-    type ValueOf<T> = T[keyof T];
-    type ObjectFit = ValueOf<Pick<React.CSSProperties, 'objectFit'>>;
+    type ObjectFit = React.CSSProperties['objectFit'];
```
Comment (guilhermefront) [inline types/react-gallery-carousel/index.d.ts:67]: I think this is just a cleaner way of achieving the same thing as before. Please let me know if I'm wrong.

File: types/react-gallery-carousel/index.d.ts [hunk 2]
```diff
@@ -111,35 +130,35 @@ declare namespace Carousel {
...
-            hasIndexBoardButtonAtMax: ButtonType;
-            hasDotButtonsAtMax: ButtonType;
-            hasCaptionsAtMax: ButtonType;
+            hasLeftButtonAtMax: SmallWidgetPositions | boolean;
+            hasRightButtonAtMax: SmallWidgetPositions | boolean;
+            hasMediaButtonAtMax: SmallWidgetPositions | boolean;
+            hasSizeButtonAtMax: SmallWidgetPositions | boolean;
+            hasIndexBoardButtonAtMax: SmallWidgetPositions | boolean;
+            hasDotButtonsAtMax: LargeWidgetPositions | boolean;
+            hasCaptionsAtMax: LargeWidgetPositions | boolean;
```
Comment (guilhermefront) [inline types/react-gallery-carousel/index.d.ts:147]: A lot of these props now has the possible positions available specified there
https://github.com/yifaneye/react-gallery-carousel#props

It's either LargeWidthPositions or SmallWidgetPositions, with a boolean also as an alternative.

File: types/react-gallery-carousel/index.d.ts [hunk 3]
```diff
@@ -111,35 +130,35 @@ declare namespace Carousel {
...
             hasThumbnailsAtMax: boolean;
             leftIcon: React.ReactNode;
             rightIcon: React.ReactNode;
             playIcon: React.ReactNode;
             pauseIcon: React.ReactNode;
-            miniIcon: React.ReactNode;
+            minIcon: React.ReactNode;
```
Comment (guilhermefront) [inline types/react-gallery-carousel/index.d.ts:153]: This fix a typo mentioned here:
https://github.com/yifaneye/react-gallery-carousel/issues/69#issuecomment-1037108269

File: types/react-gallery-carousel/index.d.ts [hunk 4]
```diff
@@ -111,35 +130,35 @@ declare namespace Carousel {
...
             maxIcon: React.ReactNode;
             activeIcon: React.ReactNode;
             passiveIcon: React.ReactNode;
             shouldSwipeOnMouse: boolean;
             shouldMaximizeOnClick: boolean;
             shouldMinimizeOnClick: boolean;
             shouldMinimizeOnSwipeDown: boolean;
-            onIndexChange(): void;
+            onIndexChange({ curIndex, curIndexForDisplay }: { curIndex: number; curIndexForDisplay: number }): void;
```
Comment (guilhermefront) [inline types/react-gallery-carousel/index.d.ts:161]: Theses params were added on this [commit](https://github.com/yifaneye/react-gallery-carousel/commit/924a06bfeb30fcb1144b5f84a7236e445bd2d965?diff=split)
image


Please let me know if this is the way that you would do it.

**Review by joshuajeschek — APPROVED** [review]
Hi, I think this is fine, thanks for the great documentation of you changes! Really helps a lot when reviewing :)

**Comment by guilhermefront** [comment]
Ready to merge

**Pull request merged**